### ISSUE TITLE
Add a method to `FileNavigator` for optionally displaying hidden files and folders.

### ISSUE THREAD

**Issue by mitchmindtree** [PistonDevelopers/conrod#746]

**Comment by Hyperchaotic** [comment]
I did some preliminary research tonight, pondering on the differences between platforms for this. On Unix like platforms (Linux, OS X) it's straightforward of course as a dot file is hidden. However on Windows we need to look for the dwFileAttributes field for the file. This is available as an [unstable API here](https://aochagavia.github.io/rust/std/os/windows/fs/trait.MetadataExt.html).

Example code:

``` rust
            use std::os::windows::fs::MetadataExt;

            const WIN_ATTR_HIDDEN: u32 = 0x2;

            let metadata = fs::metadata(&path).unwrap();
            let win_attr: u32 = metadata.file_attributes();

            print!("File {:?}, attr {:b} - ", path, win_attr);
            if (win_attr & WIN_ATTR_HIDDEN)!=0 {
                println!("is hidden! ");
            } else {
                println!("is NOT hidden! ");
            }

```
I ran the code (compiled with nightly) over a few files, one marked as hidden, and got this:
```
File "breakdance.avi", attr 10000000100000 - is NOT hidden!
File "breakdance-hidden.srt", attr 10000000100010 - is hidden!
```
The code would need to be guarded by compiler directives, [like these](https://doc.rust-lang.org/book/conditional-compilation.html) I guess.

**Comment by mitchmindtree** [comment]
This was implemented in #789, closing.
